AdyenDropIn/Components/PreApplePayComponent.swift

```diff
@@ -0,0 +1,141 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import UIKit
+import Adyen
+import AdyenComponents
+
+// :nodoc:
+internal final class PreApplePayComponent: Localizable, PresentableComponent, FinalizableComponent, PaymentComponent {
+    
+    /// :nodoc:
+    internal let paymentMethod: PaymentMethod
+    
+    /// :nodoc
+    private let applePayConfiguration: DropInComponent.ApplePayConfiguration
+    
+    /// :nodoc:
+    private let payment: Payment?
+    
+    /// :nodoc:
+    internal weak var delegate: PaymentComponentDelegate?
+    
+    /// :nodoc:
+    internal var localizationParameters: LocalizationParameters?
+    
+    /// :nodoc:
+    internal weak var presentationDelegate: PresentationDelegate?
+    
+    /// :nodoc:
+    fileprivate var applePayComponent: ApplePayComponent?
+    
+    /// :nodoc:
+    internal lazy var viewController: UIViewController = {
+        let viewController = UIViewController()
+        viewController.title = "Apple Pay"
+        let view = payment.map(\.amount)
+            .map(createModel(with:))
+            .map(PreApplePayView.init(model: ))
+        view?.delegate = self
+        viewController.view = view
+        return viewController
+    }()
+    
+    /// :nodoc:
+    internal let requiresModalPresentation: Bool = true
+    
+    /// :nodoc:
+    internal init(payment: Payment?, paymentMethod: PaymentMethod, configuration: DropInComponent.ApplePayConfiguration) {
+        self.payment = payment
+        self.paymentMethod = paymentMethod
+        self.applePayConfiguration = configuration
+    }
+    
+    /// :nodoc
+    internal func didFinalize(with success: Bool) {
+        applePayComponent?.didFinalize(with: success)
+    }
+    
+    /// :nodoc:
+    private func createModel(with amount: Payment.Amount) -> PreApplePayView.Model {
+        PreApplePayView.Model(
+            hint: amount.formatted,
+            style: PreApplePayView.Model.Style(
+                hintLabel: TextStyle(font: .preferredFont(forTextStyle: .footnote), color: UIColor.Adyen.componentSecondaryLabel),
+                backgroundColor: UIColor.Adyen.componentBackground))
+    }
+    
+    /// :nodoc:
+    private func makeViewController() -> UIViewController {
```

Comment:
You already have `lazy var viewController: UIViewController`

Reply:
👍